borg-nix-shell.el

```diff
@@ -25,31 +25,32 @@
 
 ;;; Variables:
 
-(defvar borg-nix-shell-build-use-pure-shell
-  t
-  "Determine if nix-shells should be pure.")
+(require 'borg)
+
+(defcustom borg-nix-shell-build-use-pure-shell t
+  "Determine if nix-shells should be pure."
+  :type 'boolean
+  :group 'borg)
 
 ;;; Code:
 
 (defun borg-nix-shell-borg-get (drone key &optional all)
-  "A reimplementation of `borg-get' for consistency.
+  "A reimplementation of `borg-get', which see for DRONE, KEY, and ALL.
 
 Since commit 89a939494e98056feb28da9f405e18cb140d7f22, (borg-get
 drone key nil) returns a string in some cases.
 
 This function has consistent behavior regardless of the way Borg behaves"
-(let ((ret (borg-get drone key all)))
-  (if all
-      ret
-      (if (listp ret)
-          ret
-          (list ret)))))
+  (let ((ret (borg-get drone key all)))
+    (cond (all ret)
+          ((listp ret) ret)
+          (t (list ret)))))
 
 ;;;###autoload
 (defun borg-nix-shell-build-command (drone)
-  "Return a format string for wrapping a build-step in a nix-shell.
+  "Return a format string for wrapping a build-step in a `nix-shell'.
```

Comment:
Are you certain things that arent't Lisp symbols should be quoted this way? (nix-shell is a system binary)

Reply:
nix-shell is defined in nix-shell.el in the nix-mode package. I suppose it can go either way if this should be quoted or not.